chapters/ru/chapter7/speech-to-speech.mdx

```diff
@@ -0,0 +1,255 @@
+
+На этом мы завершили первую половину нашего каскадного конвейера STST, применив на практике навыки, полученные в разделе 5, когда мы учились использовать
+модель Whisper для распознавания и перевода речи. Если вы хотите освежить в памяти какие-либо из рассмотренных нами этапов, прочтите
+раздел [Предварительно обученные модели для распознавания речи](../chapter5/asr_models) из раздела 5.
+
+## Преобразование текста в речь
+
+Вторая половина нашей каскадной системы STST связана с преобразованием английского текста в английскую речь. Для этого мы
+будем использовать предварительно обученную модель [SpeechT5 TTS](https://huggingface.co/microsoft/speecht5_tts) для английского TTS.
+🤗 В настоящее время Transformers не имеет TTS `pipeline`, поэтому нам придется использовать модель непосредственно самим.
+Ничего страшного, вы же все эксперты по использованию модели для инференса после раздела 6!
+
+Сначала загрузим процессор SpeechT5, модель и вокодер из предварительно обученной контрольной точки:
+
+```python
+from transformers import SpeechT5Processor, SpeechT5ForTextToSpeech, SpeechT5HifiGan
+
+processor = SpeechT5Processor.from_pretrained("microsoft/speecht5_tts")
+
+model = SpeechT5ForTextToSpeech.from_pretrained("microsoft/speecht5_tts")
+vocoder = SpeechT5HifiGan.from_pretrained("microsoft/speecht5_hifigan")
+```
+
+<Tip>
+    Здесь мы используем контрольную точку SpeechT5, обученную специально для английского TTS. Если вы хотите перевести
+    на язык, отличный от английского, либо замените контрольную точку на модель SpeechT5 TTS, дообученную для выбранного
+    вами языка, либо используйте контрольную точку MMS TTS, предварительно обученную для вашего целевого языка.
+</Tip>
+
+Как и в случае с моделью Whisper, модель SpeechT5 и вокодер мы поместим на GPU-ускоритель, если он у нас есть:
+
+```python
+model.to(device)
+vocoder.to(device)
+```
+
+Отлично! Давайте загрузим эмбеддинги дикторов:
+
+```python
+embeddings_dataset = load_dataset("Matthijs/cmu-arctic-xvectors", split="validation")
+speaker_embeddings = torch.tensor(embeddings_dataset[7306]["xvector"]).unsqueeze(0)
+```
+
+Теперь мы можем написать функцию, которая принимает на вход текстовый запрос и генерирует соответствующую речь. Сначала мы предварительно обработаем
+текстовый ввод с помощью процессора SpeechT5, токенизируя текст для получения входных идентификаторов. Затем мы передадим входные идентификаторы и
+эбеддинги диктора в модель SpeechT5, разместив каждый из них на ускорителе, если таковой имеется. Наконец, мы вернем сгенерированную речь обратно в
+процессор, чтобы мы могли воспроизвести ее в нашем ноутбуке ipynb:
+
+```python
+def synthesise(text):
+    inputs = processor(text=text, return_tensors="pt")
+    speech = model.generate_speech(
+        inputs["input_ids"].to(device), speaker_embeddings.to(device), vocoder=vocoder
+    )
+    return speech.cpu()
+```
+
+Let's check it works with a dummy text input:
```

Comment:
This sentence hasn't been translated.

Reply:
Added a translation of the missing sentence. Thank you for your attention Maria.